clvm/core_ops.py

```diff
@@ -54,7 +54,7 @@ def op_eq(args):
     a0 = args.first()
     a1 = args.rest().first()
     if a0.pair or a1.pair:
-        raise EvalError("= on list", args)
+        raise EvalError("= on list", a0)
```

Comment:
Aha! Here's why `args` was used: it might be `a0` is an atom, and `a1` is the list.

Reply:
Possible solution:

`raise EvalError("= on list", a0 if a0.pair else a1)`